kubemarine/thirdparties.py

```diff
@@ -38,6 +38,29 @@ def enrich_inventory_apply_upgrade_defaults(inventory, cluster):
     return inventory
 
 
+def get_thirdparty_recommended_sha(destination, cluster):
+    cluster.log.verbose("Calculate recommended sha for thirdparty %s..." % destination)
+    # Get kubeadm version
+    kubernetes_version = cluster.inventory['services']['kubeadm']['kubernetesVersion']
+    effective_kubernetes_version = ".".join(kubernetes_version.split('.')[0:2])
+
+    # Get software versions map
+    software_name = cluster.globals['thirdparties'].get(destination, {}).get('software_name', None)
+    software_versions = cluster.globals['compatibility_map']['software'].get(software_name, {})
+
+    # Return sha1 related to used kubernetes version
+    recommended_sha = software_versions[kubernetes_version].get('sha1', None) \
```

Comment:
Can this throw `KeyError: kubernetes_version` for not default 3rd-parties?

Reply:
No, because it's ternary operator and this line is used if kubernetes_version key exists in software_versions dict.
See the next line